docs/2.10.0/docs/upgrade/smart-contract-upgrades.rst

```diff
@@ -1726,6 +1726,72 @@ upgrade that dependency to a new version.
 For this reason, it is :ref:`strongly recommended that interfaces always be defined
 in their own packages separately from templates <separate_interfaces_and_exceptions>`.
 
+Developer Workflow
+------------------
+
+The following recommendations about how to set up your development environment
+for SCU will help you iterate more quickly on your projects.
+
+Multi-Package Builds
+~~~~~~~~~~~~~~~~~~~~
+
+Following the :ref:`best practices <best_practices>` outlined below in this
+document as well as the :ref:`testing recommendations <testing>` will lead to a
+proliferation of packages in your project.  Use :ref:`Multi-Package builds
+<multi-package-build>` to reliably rebuild these packages as you iterate on your
+project. Multi-package builds will also enable cross-package navigation in Daml
+Studio.
+
+The :ref:`Multi-package builds for upgrades <multi_package_upgrades>` section
+goes into more detail on how to set up multi-package builds for SCU and how it 
+can help with testing in particular.
+
+Working with a Running Canton Instance
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+With SCU, it is no longer possible to iterate on a package by
+uploading it to a running participant after each rebuild. This is because
+a participant will reject two packages with the same name and version whose content
+differs.
+
+There are two ways to work around this:
+  - restart the participant after each rebuild;
+  - change the version name of the package before each rebuild.
+
+:ref:`Environment variable interpolation <environment-variable-interpolation>`
+in the Daml Assistant can help with the latter. In the ``daml.yaml`` file of each
+of your packages, append an environment variable to the name of the package:
+
+.. code:: daml
+
+    name: my-package-v${UNIQUE_BUILD_ID}
+
+Make sure to also append the variable to the name of the DAR file produced by
+``my-package`` in the ``daml.yaml`` files that depend on it:
+
+.. code:: daml
+
+  dependencies:
+  - daml-prim
+  - daml-stdlib
+  - '../my-package/.daml/dist/my-package-${UNIQUE_BUILD_ID}.dar'
+
+Then, before invoking ``daml build --all``, increment the ``UNIQUE_BUILD_ID``
```

Comment:
Could you verify this works as well? I'm worried the caching won't notice this change

Reply:
Have verified